assets/src/components/UserSettingSnackbar.js

```diff
@@ -31,8 +37,16 @@ function UserSettingSnackbar (props) {
         vertical: 'bottom',
         horizontal: 'left'
       }}
+      // If a Snackbar requires focus to close it, then content authors should use the role of alertdialog.
+      ContentProps={{
+        'role': 'alertdialog',
+      }}
       open={savedSnackbarOpen}
-      autoHideDuration={3000}
+      // show for 200ms for each of the characters in the message
+      autoHideDuration={"{snackbarMessage}".length * 200}
```

Comment:
I think (but am not 100 percent sure) this is a bug. I believe this is not pulling in the `snackbarMessage` message string, just using the string literal, which will always be 17. You actually don't need to have `snackbarMessage` wrapped in quotation marks or curly braces; you can just use `snackbarMessage`.

I like to calculate values outside of the returned component so that I can use `console.log` more easily to see what's actually happening before it's returned. Here that would mean defining a new variable like `snackbarDuration` above `return` and then referencing it below with `autoHideDuration`.

Reply:
You're right. I'll fix it